_posts/2017/2017-08-08-why-you-should-spend-time-on-your-translation-keys.md

```diff
@@ -0,0 +1,87 @@
+---
+layout: post
+title: Why you should spend time on your translation keys
+date: "2017-08-08 09:34:02"
+image: blog-header/translation-keys.jpg
+categories: tech-stories
+authors: ["Christina"]
+---
+Translation keys are not important at all?
+Spending time on them is a waste of time?
+Let me convince you otherwise.
+
+We just finished the revision of our translation keys because we noticed the value of a good key structure.
+In the following, I will tell you why we changed the old system and what have been the most important aspects during this task.
+
+## Structure
+
+We already had functional keys before.
+Everything worked out.
+But nobody really paid attention to an overall structure or any consistency.
+For our new technology platform, we will introduce a new localization process.
+Therefore, we will work with a new tool.
+A special feature of this tool is an In-Context Editor.
+This way, translations and wording can be handled directly in the software and translators as well as UI-Writers are provided with as much context as possible.
+They know exactly where the keys are located and which kind of wording is needed.
+Thinking of the save button, you will understand why context is that important.
+*Save* is not always just *save*.
+In some cases it might be better to rename the button.
+For example, a save button at the end of a product creation process might better be called a *create product* button.
+This decision could not be made without any further information about the content of the specific page and the exact location of the button.
+
+But software might not work 24/7 and it is possible that you are forced to work without the localization tool.
+In this case you need to get the context information in a different way.
+The only source for context information will then be the keys.
+This is why the structure is that important.
+
+Let me give you an example of our new structure:
+
+{% highlight json %} {
+"loginView": {
+  "passwordTextField": {
+    "label": "Password",
+    "validationMessages": {
+      "isRequired": "Please enter a password"
+    }
+  },
+} {% endhighlight %}
+
+Here you can see the keys for one element of the log in view.
+It is a text field where you need to enter your password.
+For us, the text field is a main element with several sub-elements.
```

Comment:
Lines 51 to 56  are quite complicated to understand. Maybe easier explained with a table?